src/functions/Coverage.ps1

```diff
@@ -1043,6 +1043,229 @@ function Get-JaCoCoReportXml {
     return $xml
 }
 
+function Get-CoberturaReportXml {
+    param (
+        [parameter(Mandatory = $true)]
+        $CommandCoverage,
+        [parameter(Mandatory = $true)]
+        [object] $CoverageReport,
+        [parameter(Mandatory = $true)]
+        [long] $TotalMilliseconds
+    )
+
+    if ($null -eq $CoverageReport -or ($pester.Show -eq [Pester.OutputTypes]::None) -or $CoverageReport.NumberOfCommandsAnalyzed -eq 0) {
+        return [string]::Empty
+    }
+
+    $now = & $SafeCommands['Get-Date']
+    $nineteenSeventy = & $SafeCommands['Get-Date'] -Date "01/01/1970"
+    [long] $endTime = [math]::Floor((New-TimeSpan -start $nineteenSeventy -end $now).TotalMilliseconds)
+    [long] $startTime = [math]::Floor($endTime - $TotalMilliseconds)
+
+    $commonRoot = Get-CommonParentPath -Path $CoverageReport.AnalyzedFiles
+
+    $allLines = @()
+    foreach ($line in $CoverageReport.HitCommands) {
+        $allLines += $line
+    }
+    foreach ($line in $CoverageReport.MissedCommands) {
+        $allLines += $line
+    }
```

Comment:
```suggestion
    $allLines = [System.Collections.Generic.List[psobject]]@()
    $allLines.AddRange($CoverageReport.HitCommands)
    $allLines.AddRange($CoverageReport.MissedCommands)
```

Try something like this. `+=` copies the array over and over - very slow with potentially many lines.

Reply:
Thanks for pointing this out. I was pulling my hair out trying to figure out the best / most acceptable solution, as I originally had `@($covered, $missed) | ForEach-Object { $_ }` which was complaining about the `ForEach-Object`.